src/gui/modals/SetupDialog.cpp

```diff
@@ -1182,7 +1182,11 @@ void SetupDialog::audioInterfaceChanged(const QString & iface)
 void SetupDialog::updateBufferSizeWarning(int value)
 {
 	QString text = "<ul>";
-	if((value & (value - 1)) != 0)  // <=> value is not a power of 2 (for value > 0)
+	// 'value' is not a power of 2 (for value > 0) and under 256. On buffer sizes larger than 256
```

Comment:
Suggestion: Also comment the code `else if( m_framesPerPeriod > DEFAULT_BUFFER_SIZE )` in `src/code/AudioEngine.cpp`, because it is not clear without a comment what is actually happening there. If you remember, we were confused when we saw this the first time.

Reply:
I remember being confused slightly before opening `src/code/AudioEngine.cpp` but I'll get the idea.

How about:

```cpp
}
// if m_framesPerPeriod is larger than DEFAULT_BUFFER_SIZE, it's set to DEFAULT_BUFFER_SIZE and
// the rest is handled by an increased fifoSize.
else if( m_framesPerPeriod > DEFAULT_BUFFER_SIZE )
{
	fifoSize = m_framesPerPeriod / DEFAULT_BUFFER_SIZE;
	m_framesPerPeriod = DEFAULT_BUFFER_SIZE;
}
```
Today it's:

https://github.com/LMMS/lmms/blob/7dc00524fa263a6aebef088f8ad9eb26a4884982/src/core/AudioEngine.cpp#L128-L133

Reply:
Not sure if I would understand it by that. I think it would still help a lot if you would explain that this only affects the plugins, but not the final mixing.

Reply:
```cpp
// lmms works with chunks of size 256 and only the final mix will use the actual buffer size. Plugins
// don't see a larger buffer size than 256. if m_framesPerPeriod is larger than DEFAULT_BUFFER_SIZE,
// it's set to DEFAULT_BUFFER_SIZE and the rest is handled by an increased fifoSize.
```

Reply:
This is good in my opinion! Feel free to commit it like that.